core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/data/impl/DataFeedServiceImpl.java

```diff
@@ -142,59 +106,50 @@ public List<Resource> dataFeedToResources() {
      * {@inheritDoc}
      */
     @Override
-    public List<GridResource> dataFeedToGridResources() {
+    public List<GridResource> dataFeedToGridResources(int page) {
         try (ResourceResolver serviceResourceResolver = repositoryHelper.getServiceResourceResolver()) {
             if (serviceResourceResolver == null) {
                 LOG.warn("ResourceResolver is null, data feed to grid resources conversion is stopped");
                 return Collections.emptyList();
             }
-            return dataFeedToGridResources(serviceResourceResolver, false);
+            return dataFeedToGridResources(serviceResourceResolver, page);
         }
     }
 
-    private List<GridResource> dataFeedToGridResources(ResourceResolver resourceResolver, boolean limited) {
-        List<GridResource> gridResources = new ArrayList<>();
-        JSONArray jsonArray = JsonUtil.getJsonArrayFromFile(JSON_FEED_PATH, resourceResolver);
-        int allItemsSize = jsonArray.length();
-        if (allItemsSize > 0) {
-            int limit = limited ? Math.min(allItemsSize, UI_ITEMS_LIMIT) : allItemsSize;
-            for (int i = 0; i < limit; i++) {
-                try {
-                    JSONObject jsonObject = jsonArray.getJSONObject(i);
-                    Optional.ofNullable(JsonUtil.jsonToModel(jsonObject, GridResource.class))
-                            .ifPresent(gridResources::add);
-                } catch (JSONException e) {
-                    LOG.error("Failed to convert json object to GridResource", e);
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void modifyDataFeed(Map<String, String> propertyLocationLinkMap, int page) {
+        try (ResourceResolver serviceResourceResolver = repositoryHelper.getServiceResourceResolver()) {
+            List<GridResource> updatedResources = readGridResources(serviceResourceResolver, page).stream().peek(resource -> {
+                if (propertyLocationLinkMap.containsKey(resource.getPropertyLocation())) {
```

Comment:
❗ Please extract this code to a separate method. A lambda inside a lambda is..., ouch